includes/admin/class-wc-admin.php

```diff
@@ -203,29 +207,29 @@ public function prevent_admin_access() {
 	public function preview_emails() {
 
 		if ( isset( $_GET['preview_woocommerce_mail'] ) ) {
-			if ( ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'preview-mail' ) ) {
+			if ( ! ( isset( $_REQUEST['_wpnonce'] ) && wp_verify_nonce( sanitize_text_field( wp_unslash( $_REQUEST['_wpnonce'], 'preview-mail' ) ) ) ) ) {
 				die( 'Security check' );
 			}
 
-			// load the mailer class
+			// load the mailer class.
 			$mailer = WC()->mailer();
 
-			// get the preview email subject
+			// get the preview email subject.
 			$email_heading = __( 'HTML email template', 'woocommerce' );
 
-			// get the preview email content
+			// get the preview email content.
 			ob_start();
 			include 'views/html-email-template-preview.php';
 			$message = ob_get_clean();
 
-			// create a new email
+			// create a new email.
 			$email = new WC_Email();
 
-			// wrap the content with the email template and then add styles
+			// wrap the content with the email template and then add styles.
 			$message = apply_filters( 'woocommerce_mail_content', $email->style_inline( $mailer->wrap_message( $email_heading, $message ) ) );
 
-			// print the preview email
-			echo $message;
+			// print the preview email.
+			echo esc_html( $message );
```

Comment:
Might be html in here. Probably best to echo and add exclusion rule. It's not user generated.